src/build.rs

```diff
@@ -77,6 +77,74 @@ async fn write(doc: &fpm::FileFound, config: &fpm::Config) -> fpm::Result<()> {
             .await?;
             println!("Generated {}", file_rel_path.as_str(),);
         }
+        fpm::FileFound::MarkdownDocument(doc) => {
+            if !(doc.depth == 1 && doc.id.eq("index.md")) {
+                std::fs::create_dir_all(format!(
+                    "{}/.build/{}",
+                    doc.base_path.as_str(),
+                    doc.id.replace(".md", "")
+                ))?;
+            }
+            let file_rel_path = if doc.id.eq("index.md") {
+                "index.html".to_string()
+            } else {
+                doc.id.replace(".md", "/index.html")
+            };
+
+            let lib = fpm::Library {};
+            // Can we possibly merge the branches? Only difference is the -- ftd.text append
+            let b = match ftd::p2::Document::from(
+                &doc.id,
+                format!("-- ftd.text:\n {}", &doc.document).as_str(),
+                &lib,
+            ) {
```

Comment:
Nopes. We will need a document `FPM/markdown.ftd`, which if missing could be: 

```ftd
-- import: fpm
-- ftd.text: 

$fpm.markdown-body

```

And we need to auto include this variable: `markdown-body` and also `markdown-file`, both strings, in `fpm.ftd` file. 

We will have to ensure that the two variables are initialised correctly, like how we did in cache.rs in amitu_heroku.